app/src/main/java/com/android/sooz/filterapp/StarWarsSpeciesListAdapter.java

```diff
@@ -0,0 +1,80 @@
+package com.android.sooz.filterapp;
+
+import android.content.Intent;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.android.sooz.filterapp.models.StarWarsSpecies;
+
+import java.util.List;
+
+public class StarWarsSpeciesListAdapter extends RecyclerView.Adapter<StarWarsSpeciesListAdapter.ViewHolder> {
+
+    public class ViewHolder extends RecyclerView.ViewHolder
+            implements View.OnClickListener {
+
+        public StarWarsSpecies mAlien;
+
+        public View view;
+        public TextView name;
+        public TextView classification;
+
+        public ViewHolder(View view) {
+            super(view);
+            this.view = view;
+
+            this.name = view.findViewById(R.id.name);
+            this.classification = view.findViewById(R.id.classficiation);
+
+            view.setOnClickListener(this);
+
+        }
+
+
+        @Override
+        public void onClick(View v) {
+            Intent intent = new Intent(v.getContext(), SpeciesDetailActivity.class);
+            mAlien.fillIntent(intent);
+
+            v.getContext().startActivity(intent);
+        }
+    }
+    List<StarWarsSpecies> speciesList;
```

Comment:
Organizationally, fields are usually all defined at the top of the class, which helps people know where to look.